apcd-cms/src/client/src/components/Admin/Extensions/AdminExtensions.tsx

```diff
@@ -1,18 +1,102 @@
-import React, { useEffect, useMemo } from 'react';
-import { useExtensions, ExtensionRow } from 'hooks/admin';
+import React, { useEffect, useState } from 'react';
+import { useExtensions, ExtensionRow, ExtensionEditRow } from 'hooks/admin';
+import LoadingSpinner from 'core-components/LoadingSpinner';
+import Paginator from 'core-components/Paginator';
+import styles from './ExtensionList.module.css';
+import ViewExtensionModal from 'apcd-components/Extensions/ViewExtensionModal/ViewExtensionModal';
+import EditExtensionModal from 'apcd-components/Extensions/EditExtensionModal/EditExtensionModal';
 
 export const AdminExtensions: React.FC = () => {
-  const { data, isLoading, isError } = useExtensions({});
+  const [status, setStatus] = useState('All');
+  const [org, setOrg] = useState('All');
+  const [page, setPage] = useState(1);
+  const { data, isLoading, isError, refetch } = useExtensions(status, org, page);
+  const [isViewModalOpen, setIsViewModalOpen] = useState(false);
+  const [isEditModalOpen, setIsEditModalOpen] = useState(false);
 
-  if (isLoading) {
+    const [selectedExtension, setSelectedExtension] =
+    useState<ExtensionRow | null>(null);
+
+  const clearSelections = () => {
+    setStatus('');
+    setOrg('');
+    setPage(1);
+  };
+
+  useEffect(() => {
+    refetch();
+  }, [status, org, page, refetch]);
+
+   if (isLoading) {
     return <div>Loading...</div>;
   }
 
   if (isError) {
     return <div>Error loading data</div>;
   }
 
+  const openAction = (
+    event: React.ChangeEvent<HTMLSelectElement>,
+    ext_id: string
+  ) => {
+    const actionsDropdown = event.target;
+    const selectedOption = actionsDropdown.value;
+    setSelectedExtension(
+      data?.page.find((x) => x.ext_id === ext_id) ?? null
+    );
+    if (selectedOption == 'viewExtension') {
+      setIsViewModalOpen(true);
+    } else if (selectedOption == 'editExtension') {
+        setIsEditModalOpen(true);
+    }
+    actionsDropdown.selectedIndex = 0;
+  };
+
+
   return (
+      <div>
+         <div className="filter-container">
+        <div className="filter-content">
+          {/* Filter */}
+          <div style={{ display: 'flex', alignItems: 'center', gap: '10px' }}>
```

Comment:
```suggestion
          <div className={styles.filterRow}>
```
Then create .filterRow in your ExtensionList.module.css like
```
.filterRow {
   display: flex;
   align-items: center;
   gap: 10px;
}